ext/xmlreader/php_xmlreader.c

```diff
@@ -781,8 +790,7 @@ PHP_METHOD(XMLReader, read)
 		}
 	}
 
-	php_error_docref(NULL, E_WARNING, "Load Data before trying to read");
-	RETURN_FALSE;
+	zend_throw_error(NULL, "Data must be loaded before reading");
```

Comment:
Style nit: Rewrite this as early return?

Reply:
Makes sense for me!